### PR TITLE
Get mac address with vm cli command

### PR DESCRIPTION
**Pull request by tinawang123**
Get mac address with vm cli command

- Related ticket: https://progress.opensuse.org/issues/164742
- Verification run:  https://openqa.suse.de/tests/15110957

### CODE REVIEW

**Comment by nanzhg** [comment]
LGTM, please have a test run for this PR.

File: lib/virt_autotest/esxi_utils.pm [hunk 1]
```diff
@@ -104,6 +105,18 @@ sub esxi_vm_public_ip {
...
 
+sub esxi_vm_mac_address {
+    my $vmid = shift;
+    my $vim_cmd;
+    my $vm_mac;
+
+    if (is_svirt) {
+        $vim_cmd = qq(vim-cmd vmsvc/get.guest $vmid | grep macAddress | sed -n 1p | cut -d'\"' -f2);
+        (undef, $vm_mac) = console('svirt')->run_cmd($vim_cmd, domain => 'sshVMwareServer', wantarray => 1);
+    }
```
Comment (nanzhg) [inline lib/virt_autotest/esxi_utils.pm:116]: Could you also support qemu backend in this new function?

File: tests/microos/networkmanager.pm [hunk 2]
```diff
@@ -65,9 +67,16 @@ sub run {
         'mac_addr' => get_var('NICMAC'),
         'local_ip' => '10.0.2.15'
     );
-    %net_config = is_s390x ? %network_s390x : %network_qemu;
-    $nic_name = script_output("grep $net_config{'mac_addr'} /sys/class/net/*/address |cut -d / -f 5");
-
+    %net_config = is_svirt ? %network_s390x : %network_qemu;
+    if (is_s390x || is_qemu) {
+        $nic_name = script_output("grep $net_config{'mac_addr'} /sys/class/net/*/address |cut -d / -f 5");
+    } elsif (is_svirt_except_s390x) {
```
Comment (mloviska) [inline tests/microos/networkmanager.pm:73]: This should directly target vmware only and not a broader group of cases.

Comment (rfan1) [reply tests/microos/networkmanager.pm]: I agree with @mloviska on this point.

File: tests/microos/networkmanager.pm [hunk 3]
```diff
@@ -65,7 +67,25 @@ sub run {
         'mac_addr' => get_var('NICMAC'),
         'local_ip' => '10.0.2.15'
     );
-    %net_config = is_s390x ? %network_s390x : %network_qemu;
+    # the below network confiration is used on svirt except s390x
+    %network_svirt = ();
+    if (is_svirt_except_s390x) {
```
Comment (rfan1) [inline tests/microos/networkmanager.pm:72]:
```suggestion
    if (is_vmware) {
```

Comment (tinawang123) [reply tests/microos/networkmanager.pm]: Updated

File: tests/microos/networkmanager.pm [hunk 4]
```diff
@@ -108,16 +129,18 @@ true'
...
-    assert_script_run('ln -rsf /run/systemd/resolve/stub-resolv.conf /etc/resolv.conf');
-    systemctl('restart NetworkManager');
-    record_info('with systemd-resolved');
-    # DNS-Manager check
-    ($RcManager, $mode) = dns_mgr();
-    die 'wrong DNS-Manager is currently used for systemd-resolved' if ($RcManager !~ /unmanaged/ || $mode !~ /systemd-resolved/);
-    ping_check;
-    # Restore the original system config
-    restore_config;
+    unless (is_vmware) {
```
Comment (rfan1) [inline tests/microos/networkmanager.pm:132]: @tinawang123 why vmware is skipped here?

Comment (tinawang123) [reply tests/microos/networkmanager.pm:132]: VMware vm system doesn't have '/run/systemd/resolve/stub-resolv.conf'

**Review by rfan1 — APPROVED** [review]

**Pull request merged**